spec/reservation_mgr_spec.rb

```diff
@@ -0,0 +1,157 @@
+require_relative 'spec_helper'
+
+describe "ReservationMgr Test" do
+  describe "initialize" do
+    before do
+      @new_ResMgr = ReservationMgr.new(3)
+    end
+    it "has an array of Rooms that 3 rooms long" do
+      expect(@new_ResMgr.rooms).must_be_kind_of Array
+      expect(@new_ResMgr.rooms.length).must_equal 3
+    end
+    it "has instances of Rooms at each index" do
+      expect(@new_ResMgr.rooms[0]).must_be_kind_of Room
+    end
+    it "has an empty array of reservations at each index" do
+      expect(@new_ResMgr.reservations).must_equal []
+    end
+  end
+
+  describe "make_reservation method" do
+    before do
+      @new_ResMgr = ReservationMgr.new(3)
+    end
+    it "stores a new instance of a reservation if there are rooms available " do
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      expect(@new_ResMgr.reservations[0]).must_be_kind_of Reservation
+    end
+    it "returns and argument error if there are no rooms available" do
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      expect{@new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))}.must_raise ArgumentError
+    end
+    it "will create two new reservations if someone wants two rooms" do
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"),rooms: 2)
+      expect(@new_ResMgr.reservations.length).must_equal 2
+    end
+    it "will raise an argument error if not enough rooms are available" do
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      expect{@new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"),rooms: 2)}.must_raise ArgumentError
+    end
+    it "will not reserve a room on the same day if the dates are inside the date range of a previous reservation" do
+      @new_ResMgr.make_reservation(Date.parse("2018-09-04"),Date.parse("2018-09-08"))
+      @new_ResMgr.make_reservation(Date.parse("2018-09-05"),Date.parse("2018-09-07"))
+      expect(@new_ResMgr.reservations.length).must_equal 2
+      expect(@new_ResMgr.rooms[0].unavailable_dates.length).must_equal 1
+      expect(@new_ResMgr.rooms[1].unavailable_dates.length).must_equal 1
+    end
+    it "will add two instances of res to one room if they are not on the same date" do
```

Comment:
You are also making a reservation that begins on the date the previous one ended.  I would be specific that you are testing that feature.